flocker/control/test/test_httpapi.py

```diff
@@ -456,3 +460,259 @@ def test_listens_endpoint(self):
         port = server[0]
         factory = server[1].__class__
         self.assertEqual((port, factory), (6789, Site))
+
+
+class DatasetsStateTestsMixin(APITestsMixin):
+    """
+    Tests for the service version description endpoint at ``/state/datasets``.
+    """
+    def test_empty(self):
+        """
+        Initial state is no datasets
+        """
+        response = []
+        return self.assertResult(
+            b"GET", b"/state/datasets", None, OK, response
+        )
+
+    def test_one_dataset(self):
+        """
+        One dataset
+        """
+        expected_dataset = Dataset(dataset_id=unicode(uuid4()))
+        expected_manifestation = Manifestation(
+            dataset=expected_dataset, primary=True)
+        expected_hostname = u"192.0.2.101"
+        self.cluster_state_service.update_node_state(
+            expected_hostname, NodeState(
+                running=[],
+                not_running=[],
+                other_manifestations=frozenset([expected_manifestation])
+            )
+        )
+        expected_dict = dict(
+            dataset_id=expected_dataset.dataset_id,
+            primary=expected_hostname,
+            metadata={}
+        )
+        response = [expected_dict]
+        return self.assertResult(
+            b"GET", b"/state/datasets", None, OK, response
+        )
```

Comment:
Done. Added an `assertItemsEqual` version of `assertResult` to compare the unordered list response correctly.